### PR TITLE
test promotion flag

### PR DESCRIPTION
**Pull request by FissoreD**

### CODE REVIEW

File: tests/suite/dune [hunk 1]
```diff
@@ -1,6 +1,6 @@
 (library
   (name test_suite)
-  (libraries str unix)
+  (libraries str unix fileutils)
```
Comment (gares) [inline tests/suite/dune:3]: ?

File: tests/suite/elpi_api.ml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 (* ------------------------------------------------------------------------- *)
 open Suite
 
-let declare = Test.declare ~category:(Filename.(chop_extension (basename __FILE__)))
+let declare = Test.declare ~category:(Filename.(chop_extension (basename __FILE__))) ~promote:false
```
Comment (gares) [inline tests/suite/elpi_api.ml:6]: I home `Test.declare` can have an optional argument `?(promote=false)` of type `?promote:bool` and avoid the long diff

File: tests/suite/suite.ml [hunk 3]
```diff
@@ -411,6 +412,9 @@ let () = Runner.declare
         | Test.Success -> Runner.Success { walltime; typechecking; execution; mem }
         | Test.SuccessOutput rex when Util.with_log log (match_rex rex) -> Runner.Success { walltime; typechecking; execution; mem }
         | Test.SuccessOutputFile { sample; adjust; reference } when match_file ~log sample adjust (sources^"/"^reference) -> Runner.Success { walltime; typechecking; execution; mem }
+        | Test.SuccessOutputFile { sample; adjust; reference } when promote ->
+            FileUtil.cp [adjust sample] (sources^"/"^reference);
```
Comment (gares) [inline tests/suite/suite.ml:416]: `Sys.rename` should work here

File: tests/suite/suite.mli [hunk 4]
```diff
@@ -68,7 +69,7 @@ type time = {
   walltime : float;
   mem : int;
 }
-type rc = Timeout of float | Success of time | Failure of time
+type rc = Timeout of float | Success of time | Failure of time | Promote of time
```
Comment (gares) [inline tests/suite/suite.mli:72]: `| Promote of bool`

**Comment by gares** [comment]
Thanks, this is useful!

File: tests/suite/dune (hunk 1 above)
Comment (FissoreD) [reply tests/suite/dune:3]: `Sys.command("cp source dest")` is not portable on system like windows I think (?)
Fileutils should implement system commands like cp/rm etc... that are POSIX compatible

File: tests/suite/suite.mli (hunk 4 above)
Comment (gares) [inline tests/suite/suite.mli:72]:
```suggestion
type rc = Timeout of float | Success of time | Failure of time | Promote of bool
```

File: tests/suite/suite.mli [hunk 5]
```diff
@@ -68,7 +69,7 @@ type time = {
   walltime : float;
   mem : int;
 }
-type rc = Timeout of float | Success of time | Failure of time
+type rc = Timeout of float | Success of time | Failure of time | Promote of bool
```
Comment (gares) [inline tests/suite/suite.mli:72]:
```suggestion
type rc = Timeout of float | Success of time | Failure of time | Promote of time
```

**Pull request merged**